runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -116,6 +116,80 @@ void J9::RecognizedCallTransformer::process_java_lang_StrictMath_and_Math_sqrt(T
 
    TR::TransformUtil::removeTree(comp(), treetop);
    }
+
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+/*
+Transforms calls to java/lang/invoke/Invokers.checkExactType to the more performant ZEROCHK.
+
+Blocks before transformation: ==>
+
+start Block_A
+...
+treetop
+        call  java/lang/invoke/Invokers.checkExactType(Ljava/lang/invoke/MethodHandle;Ljava/lang/invoke/MethodType;)
+           ==>aload
+           ==>aload
+...
+end Block_A
+
+Blocks after transformation: ==>
+
+start Block_A
+...
+treetop
+        ZEROCHK
+          acmpeq
+             ==>aload <expected type>
+             aloadi <MethodHandle.type>
+               ==>aload <MethodHandle>
+...
+end Block_A
+
+*/
+void J9::RecognizedCallTransformer::process_java_lang_invoke_Invokers_checkExactType(TR::TreeTop* treetop, TR::Node* node)
+   {
+   TR::Node * methodHandleNode = node->getChild(0);
+   TR::Node * expectedTypeNode = node->getChild(1);
```

Comment:
It's better to use `getArgument`. The child index may not match argument index for some methods